R/mean-model.R

```diff
@@ -0,0 +1,250 @@
+  "MeanModel",
+  public = list(
+    modelFilePath = NULL, #a path to a pkml file that will be loaded using loadSimulation()
+    simModel = NULL, #a model previously loaded using loadSimulation, an alternative to providing modelFilePath
+    modelDisplayName = NULL, #a string, to be set to name of model file if not supplied
+    simulationOutputs = NULL,
+    massBalance = NULL,
+    PKAnalysis = NULL,
+    additionalPKAnalysis = NULL,
+    #OPTIONAL Name of the user-defined function for calculation of "non-standard" PKParameters (s. #23)
+    simulationOutputsFolder = NULL, #a string, Subfolder for calculation outputs
+    #Data display name for report
+    #Time raster
+
+
+    simulationResults = NULL,
+    sensitivityResults = NULL,
+
+
+    initialize = function(modelFilePath = NULL,
+                          simModel = NULL,
+                          modelDisplayName = NULL,
+                          simulationOutputsFolder = "." #default output folder is current directory
+    ) {
+
+      if (!is.null(modelFilePath)){
+        self$modelFilePath <- modelFilePath
+      } else{
+
+        if (!is.null(simModel)){
+          self$modelFilePath <- simModel$sourceFile
+        } else{
+          stop("Either modelFilePath or simModel need to be provided.")
+        }
+
+      }
+
+      self$modelDisplayName <- modelDisplayName
+      self$simulationOutputsFolder <- simulationOutputsFolder
+
+
+
+      #Active task list: (ie tasks that are to be started when workflow begins)
+
+
+    },
+
+    #TASKS
+    simulateMeanModel = function(saveSimulation = FALSE) {
+
+      loadedModel <- loadSimulation(self$modelFilePath, loadFromCache = FALSE)
+      # if (calculateMassBalance) {  #If statement to add all outputs to loadedModel's output list if a mass balance is required
+      #   compoundList <- getCompoundsList(loadedModel) #Get a vector of all compound names
+      #   pathsPerCompoundList <-list()
+      #   for (cmp in compoundList){ #For loop to get the list of paths for each compound
+      #     pathsPerCompoundList[[cmp]]  <- getPathsForMoleculeAmount(simulation = loadedModel,cmp)
+      #     addOutputs(quantitiesOrPaths = pathsPerCompoundList[[cmp]], simulation = loadedModel) #add outputs for current compound to loadedModel's output list
+      #   }
+      # }
+
+      #Simulate mean model (details s. #31)
+      #Remarks: Mass balance plots require all model outputs, thus it's not enough to save only model outputs of interest
+
+      # IF TIME RASTER PROVIDED THEN USE IT, ELSE USE DEFAULT SIMULATION TIMES IN PKML
+      self$simulationResults <- runSimulation(loadedModel)
+      self$simulationOutputs <- getOutputValuesTLF(simulationResults = self$simulationResults, quantitiesOrPaths = self$modelOutputsList , individualIds = 0 , population =  loadPopulation(csvPopulationFile = "./data/popData.csv"))  #using this placeholder population CSV file for now...need to make optional the population input to getOutputValuesTLF
```

Comment:
getOutputValuesTLF does not exist anymore in ospsuite.core. This needs to be updated